ovis_jog_control/scripts/ovis_control.py

```diff
@@ -0,0 +1,465 @@
+
+        self.cmd_scalling()
+
+    
+    def set_cmd_from_joy(self, data):
+        """ Set cmd from joy msg
+        """
+
+        self.position_mode = True if data.axes[2] == -1 else False
+        self.orientation_mode = True if data.axes[5] == -1 else False
+
+        if self.position_mode and self.orientation_mode:
+            return False
+
+        if self.position_mode:
+            self.cmd_x = data.axes[1]    # Up/Down Axis stick left
+            self.cmd_y = -data.axes[0]   # Left/Right Axis stick left
+            self.cmd_z = data.axes[4]    # Up/Down Axis stick right
+            self.cmd_roll = 0
+            self.cmd_pitch = 0
+            self.cmd_yaw = 0
+            self.cmd_scalling()
+            return True
+
+        if self.orientation_mode:
+            self.cmd_x = 0
+            self.cmd_y = 0
+            self.cmd_z = 0
+            self.cmd_roll = data.axes[1]    # Up/Down Axis stick left
+            self.cmd_pitch = data.axes[0]   # Left/Right  Axis stick left
+            self.cmd_yaw = data.axes[4]     # Up/Down Axis stick right
+            self.cmd_scalling()
+            return True
+
+        if data.buttons[7]: # Start
+            self.sleep_rate.sleep()
+            self.toggle_local_world_ref = True
+            return False
+
+        if data.buttons[3]: # Y
+            self.sleep_rate.sleep()
+            self.toggle_end_effector = True
+            return False
+
+        self.toggle_end_effector = False
+        self.toggle_local_world_ref = False
+
+        return False
+
+    def cmd_scalling(self):
+        """ 
+        """
+        self.cmd_x *= self.scaling
+        self.cmd_y *= self.scaling
+        self.cmd_z *= self.scaling
+        self.cmd_roll *= self.scaling
+        self.cmd_pitch *= self.scaling
+        self.cmd_yaw *= self.scaling 
+
+    def print_cmd(self):
+        """ Print useful informations (x, y, z, roll, pitch, yaw, ref)
+        """
+    
+        print("\n*******************************************")
+        print("x: {0:4}, Roll: {1:4}".format(self.cmd_x,self.cmd_roll))
+        print("y: {0:4}, pitch:{1:4}".format(self.cmd_y,self.cmd_pitch))
+        print("z: {0:4}, Yaw:  {1:4}".format(self.cmd_z,self.cmd_yaw))
+        print("Current RPY:  {0}".format(self.group.get_current_rpy()))
+        print("Reference: {0}".format("Local" if self.local_ref else "World"))
+        print("End effector: {0}".format(self.group.get_end_effector_link()))
+        print("Pose Ref Frame: {0}".format(self.group.get_pose_reference_frame()))
+        print("*******************************************\n")
+        print("JACOBIAN {0}".format(self.group.get_jacobian_matrix(self.group.get_current_joint_values()))) 
+        print("*******************************************\n")
+
+
+
+if __name__ == '__main__':
+    try:
+        my_planner = Commander()
+
+    except rospy.ROSInterruptException:
+        pass
+
+    moveit_commander.roscpp_shutdown()
```

Comment:
For my own knowledge, why is roscpp shutdown here?

Reply:
To shutdown moveit_commander/moveit node cleanly. A cleanup function with safe arm position and some loginfo could be added.